### PR TITLE
HTTP Part 0

### PR DESCRIPTION
**Pull request by ortyomka**
Issue #597 

### STD Driver Extension
STD driver haven't depended on specific implementation.
Interface will be chosen depends on `Options` in connect stage.

### New connection
HTTP connection was added with minimum functionality (Open, Ping, Close).
Other functions return `ErrHttpNotSupported` for now. Their implementation will be in the next PRs.

### Tests
Added tests for HTTP connection through STD interface.

### Breaking changes
None. All tests passed successfully.

cc @gingerwizard

### CODE REVIEW

**Comment by gingerwizard** [comment]
This looks extremely promising and an elegant solution @ortyomka Tests are failing as we need to update the container to expose http.

**Comment by gingerwizard** [comment]
See https://github.com/ClickHouse/clickhouse-go/blob/v2/.github/workflows/run-tests.yml#L34

**Comment by ortyomka** [comment]
I have exposed HTTP port for tests

Could you run the tests again? 
@gingerwizard

File: clickhouse_std.go [hunk 1]
```diff
@@ -93,9 +118,21 @@ func OpenDB(opt *Options) *sql.DB {
 	})
 }
 
+type layer interface {
```
Comment (gingerwizard) [inline clickhouse_std.go:121]: can we call this `transport`?

File: clickhouse_std.go [hunk 2]
```diff
@@ -107,18 +144,77 @@ func (d *stdDriver) Open(dsn string) (_ driver.Conn, err error) {
...
+
+func (std *nativeLayer) ResetSession(ctx context.Context) error {
 	if std.conn.isBad() {
 		return driver.ErrBadConn
 	}
 	return nil
 }
 
-func (std *stdDriver) Ping(ctx context.Context) error { return std.conn.ping(ctx) }
+func (std *nativeLayer) Ping(ctx context.Context) error { return std.conn.ping(ctx) }
```
Comment (gingerwizard) [inline clickhouse_std.go:213]: likewise `nativeTransport`

File: clickhouse_std.go [hunk 3]
```diff
@@ -107,18 +144,77 @@ func (d *stdDriver) Open(dsn string) (_ driver.Conn, err error) {
 	return (&stdConnOpener{opt: &opt}).Connect(context.Background())
 }
 
-func (std *stdDriver) ResetSession(ctx context.Context) error {
+var ErrHttpNotSupported = errors.New("HTTP: not supported")
+
+type httpLayer struct {
```
Comment (gingerwizard) [inline clickhouse_std.go:149]: see above, maybe `httpTransport`

**Review by gingerwizard — CHANGES_REQUESTED** [review]
Minor naming changes. This LGTM - i prefer `transport` over `layer` if possible.

**Comment by gingerwizard** [comment]
@ortyomka minor naming change otherwise happy to merge ASAP.

**Comment by ortyomka** [comment]
@gingerwizard, done

**Comment by gingerwizard** [comment]
Great start, thanks @ortyomka

**Pull request merged**